src/services/findAllReferences.ts

```diff
@@ -840,6 +840,14 @@ namespace ts.FindAllReferences {
                     // import("foo") with no qualifier will reference the `export =` of the module, which may be referenced anyway.
                     return nodeEntry(reference.literal);
                 }
+                else if (reference.kind === "implicit") {
+                    // Return either: The first JSX node in the file, the first statement of the file, or the whole file if neither of those exist
+                    const range = forEachChildRecursively(
+                        reference.referencingFile,
+                        n => isJsxElement(n) || isJsxSelfClosingElement(n) || isJsxFragment(n) ? n : undefined
+                    ) || reference.referencingFile.statements[0] || reference.referencingFile;

```

Comment:
I thought a bit more about the `statements[0]` case. It would be very weird if the first statement in your file is an unrelated import...

Maybe we need to find the first statement that's not an import or an export with a module specifier?

Reply:
Not sure it matters - `import "./thing"` being the ref is probably just as odd as `var x = 1`. Point is really that the file itself is the reference, we just don't want to say "the whole file is the reference" because it'd be a really really big and useless usage highlight span.